### PR TITLE
Update dependencies to latest consumable version

### PR DESCRIPTION
**Pull request by dsimansk**
Fixes JIRA #

## Proposed Changes

- Pin k8s.io deps to latest 0.29
  - Pinning kube should enable future Dependabot updates without undesired version increments
- Update operator-framework/api to latest
- Update prometheus/client to latest release on go 1.22
  - We can't go with newer version, because it requires go 1.23 already. 

Incorporates Dependabot PRs: https://github.com/openshift-knative/serverless-operator/pulls?q=is:pr+is:open+label:dependencies
I'll close those PRs once this one is merged eventually.

/cc @skonto @ReToCode @pierDipi

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -190,7 +191,18 @@ replace (
 	github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20240510131258-f646d5f29250
 )
 
-// This replacement can be removed once we bump to k8s.io@v1.30.x, as from release 1.30+
-// https://github.com/kubernetes/code-generator/blob/release-1.30/go.mod, code-generator will use
-// the new v2 version of kube-openapi.
-replace k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20231010175941-2dd684a91f00
+replace (
```
Comment (ReToCode) [inline go.mod:194]: maybe add a TODO that we remove this for 1.16/1.36?

**Review by ReToCode — APPROVED** [review]
LGTM other than the nit.

File: go.mod (hunk 1 above)
Comment (dsimansk) [reply go.mod:194]: Yep, that's a great idea. Done!

**Review by pierDipi — APPROVED** [review]
LGTM

**Pull request merged**